### ISSUE TITLE
Box "between" border places borders at end of gaps on mobile viewports

### ISSUE THREAD

**Issue by myktra** [grommet/grommet#3788]
The new *between* value for the `border` prop on boxes aims to place borders between items in a box, centered within the space specified by the `gap` prop. This is great! However, because of the way the border is specified in CSS as fixed pixel distance (rather than a percentage?) it can appear at the *end* of a gap on mobile viewports.

### Expected Behavior

The default behavior (arguably?) ought to be that the border remains placed in the *middle* of the gap.

### Actual Behavior

The border is placed at the end of the gap.

### URL, screen shot, or Codepen exhibiting the issue

![2020-02-26 09 58 14](https://user-images.githubusercontent.com/653521/75356970-a2486b00-587e-11ea-9072-cca6f77075af.gif)

### Steps to Reproduce

Use this code:

```jsx
<Box direction="column" gap="medium" border="between">
  <Box background="light-2">
    <Text>Hi!</Text>
  </Box>
  <Box background="light-2">
    <Text>Hi!</Text>
  </Box>
</Box>
```

Notice that when the viewport is shrunk, the gap decreases in height by about 50%. Because the `top` attribute of the CSS rule is set to a fixed `23.5px` (likely a formula based on spacings specified by the theme), the border appears at the end of the gap. If the `top` attribute of the CSS rule were set to `50%` it would appear in the middle of the gap as expected.

### Your Environment

- Grommet version: 2.11.1
- Browser Name and version: Chrome 79
- Operating System and version (desktop or mobile): macOS 10.15

**Comment by ShimiSun** [comment]
Thanks for reporting buddy! I'm glad you are using a recent release and giving this new prop-value a try, looks like the issue starts here https://github.com/grommet/grommet/blob/8fbdf884f339944c5bbf94d6bbc99c228b7af386/src/js/components/Box/StyledBox.js#L589
Feel free to start a conversation with filing a potential PR fix.

**Comment by myktra** [comment]
Thanks back! Right, it would appear in this code that for some reason we're not using the responsive gap/edge size, as we otherwise are elsewhere in this block.

See: https://github.com/grommet/grommet/blob/8fbdf884f339944c5bbf94d6bbc99c228b7af386/src/js/components/Box/StyledBox.js#L580

This would apply to the `top` and `left` attributes depending on the box `direction` prop:

https://github.com/grommet/grommet/blob/8fbdf884f339944c5bbf94d6bbc99c228b7af386/src/js/components/Box/StyledBox.js#L600
https://github.com/grommet/grommet/blob/8fbdf884f339944c5bbf94d6bbc99c228b7af386/src/js/components/Box/StyledBox.js#L631

**Comment by ShimiSun** [comment]
Right, feel like filing a PR?

**Comment by myktra** [comment]
Hi, not sure this is completely resolved, at least not in the latest 2.12.0 drop.

Have a peek at these demos - resize the viewport and notice the border between the two items isn't properly aligned when the gap shrinks.

https://codesandbox.io/s/cocky-poincare-lus8d
https://storybook.grommet.io/?path=/story/box--border (look at the "between column" area)

It could be that only this is only an issue for borders between elements in a Box with `direction="column"`.

HTH!
